spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/util/KeyUtil.java

```diff
@@ -46,6 +46,7 @@ public static Key getKeyWithoutAncestors(Key entityKey) {
           Key.newBuilder(entityKey.getProjectId(), entityKey.getKind(), entityKey.getId());
     }
     ancestorLookupKey.setNamespace(entityKey.getNamespace());
+    ancestorLookupKey.setDatabaseId(entityKey.getDatabaseId());
```

Comment:
It's too bad the ancestors can't be explicitly cleared in the builder, or we'd be able to use the copy constructors to avoid direct knowledge of all the individual fields.
```java
public static Key getKeyWithoutAncestors(Key entityKey) {
  return Key.newBuilder(entityKey).withoutAncestors().build();
}